dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/AbstractProxyProtocol.java

```diff
@@ -67,7 +67,11 @@ public void setProxyFactory(ProxyFactory proxyFactory) {
         Exporter<T> exporter = (Exporter<T>) exporterMap.get(uri);
         if (exporter != null) {
             // When modifying the configuration through override, you need to re-expose the newly modified service.
-            if (exporter.getInvoker().getUrl().equals(invoker.getUrl())) {
+            if ("hessian".equals(invoker.getUrl().getProtocol())) {
+                if (exporter.getInvoker().getUrl().equals(invoker.getUrl())) {

```

Comment:
Could `getUrl` return null here? If so, `Objects.equals` might be better.

Reply:
You are right, I will modify it.